src/state/slices/tradeQuoteSlice/helpers.ts

```diff
@@ -123,31 +148,113 @@ export const getTotalProtocolFeeByAsset = (
+      if (aTime === undefined) return 1 // Push undefined to last, we don't know the ETA here
+      if (bTime === undefined) return -1 // Keep defined ETA above undefined
+
+      // Number compare is safe since we're dealing with unix timestamps
+      return aTime - bTime
+    })
+    return sorted
+  }
+  const iteratees: ((quote: ApiQuote) => any)[] = (() => {
+    switch (sortOption) {
+      case QuoteSortOption.LOWEST_GAS:
+        return [
+          // Presort by un/available network fees
+          (quote: ApiQuote) => {
+            if (!quote.quote?.steps) return true
+
+            // Unknown gas - assume latest
+            if (quote.quote.steps.every(step => !step?.feeData?.networkFeeCryptoBaseUnit))
+              return true
+
+            return false
+          },
+          // Then sort by the actual fee amount in user currency
+          (quote: ApiQuote) => {
+            return getNetworkFeeUserCurrency(quote.quote)
+          },
+        ]
+      case QuoteSortOption.BEST_RATE:
+      default:
+        return [
+          (quote: ApiQuote) => {
+            if (!quote.quote?.steps?.length) return bn(0)
+
+            // Get the last step for multi-hop trades
+            const steps = quote.quote.steps
+            const lastStep = steps[steps.length - 1]
+
+            // Use buyAmountAfterFeesCryptoBaseUnit which should match the displayed amount
+            const buyAmount = bnOrZero(lastStep.buyAmountAfterFeesCryptoBaseUnit)
+
+            return buyAmount
+          },
+        ]
+    }
+  })()
+
+  const sortOrders: ('asc' | 'desc')[] = (() => {
+    switch (sortOption) {
+      case QuoteSortOption.LOWEST_GAS:
+        return ['asc', 'asc'] // Lowest to highest network fees
+      case QuoteSortOption.BEST_RATE:
+      default:
+        return ['desc'] // Highest to lowest received amount after fees
+    }
+  })()
+
+  const ordered = orderBy(quotesWithoutErrors, iteratees, sortOrders)
+
+  return ordered
 }
 
 export const sortTradeQuotes = (
   tradeQuotes: PartialRecord<SwapperName, Record<string, ApiQuote>>,
+  sortOption: QuoteSortOption = QuoteSortOption.BEST_RATE,
 ): ApiQuote[] => {
   const allQuotes = Object.values(tradeQuotes)
     .filter(isSome)
     .map(swapperQuotes => Object.values(swapperQuotes))
     .flat()
-  const happyQuotes = sortApiQuotes(allQuotes.filter(({ errors }) => errors.length === 0))
-  const errorQuotes = sortApiQuotes(allQuotes.filter(({ errors }) => errors.length > 0))
-  return [...happyQuotes, ...errorQuotes]
+
+  // Split quotes into those with and without errors
+  const quotesWithoutErrors = allQuotes.filter(quote => quote.errors.length === 0)
+  const quotesWithErrors = allQuotes.filter(quote => quote.errors.length > 0)
+
+  // Only sort quotes without errors
+  const sortedHappyQuotes = sortApiQuotes(quotesWithoutErrors, sortOption)
+
+  // Return sorted quotes without errors first, then quotes with errors
+  return [...sortedHappyQuotes, ...quotesWithErrors]
```

Comment:
Is this really needed? As they are separated anyway, it feels a bit too much for almost nothing

But happy with is as your 🧠 would tell me "this bit doesn't know about this particularity of our UI"

Reply:
It's a bit of a confusing one and I was just as confused by it but yeah this is still needed!
There's kind of like two layers of errors:
- "hard errors" meaning there was no quote gotten whatsoever, these go to "Unavailable"
- "soft errors" (there was a quote but it's not actionable e.g not enough fundus, these gt pushed at the end of the quotes list because they can't be actioned

Reply:
Ahh, makes sense! Thanks for the details